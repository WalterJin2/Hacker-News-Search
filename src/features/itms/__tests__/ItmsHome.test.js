import { rest } from "msw";
import { screen } from "@testing-library/react";
import { server } from "../../../test/server";
import { renderWithProviders } from "../../../test/test-utils";
import ItemHome from "../ItmsHome";
  it("ItemHome renders correctly", async () => {
    // force msw to return error response
    server.use(
      rest.get(
        "/search?query=&tags=(story,comment)&hitsPerPage=20",
        (req, res, ctx) => {
            return res(ctx.status(200), ctx.json({}));
        }
      )
    );


  const { asFragment } = renderWithProviders(<ItemHome  />);
  expect(asFragment()).toMatchSnapshot();

  });

// it("Search renders correctly", async () => {
//   apiMock
//     .get("/search?query=&tags=(story,comment)&hitsPerPage=20")
//     .reply(200, {});

//   const component = await renderComponent(<Search />, getInitialState());
//   const result = component.toJSON();
//   expect(result).toMatchSnapshot();
// });
