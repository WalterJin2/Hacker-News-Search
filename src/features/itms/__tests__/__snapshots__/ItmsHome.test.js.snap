// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemHome renders correctly 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col justify-between items-center w-full grow px-6 py-2  "
  >
    <div
      class="w-full"
    >
      <label
        class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-gray-300"
      >
        Search
      </label>
      <div
        class="relative"
      >
        <div
          class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5 text-gray-500 dark:text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <input
          class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
          id="default-search"
          placeholder="Search..."
          required=""
          type="search"
        />
        <button
          class="text-white absolute right-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
    <div
      class="flex justify-center items-center  w-full grow"
    >
      Loading...
      <span
        style="display: inherit;"
      >
        <span
          style="background-color: rgb(0, 0, 0); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.12s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
        />
        <span
          style="background-color: rgb(0, 0, 0); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.24s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
        />
        <span
          style="background-color: rgb(0, 0, 0); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.36s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
